app/src/main/java/com/mealsmadeeasy/ui/splash/SplashActivity.kt

```diff
@@ -1,15 +1,122 @@
+                .requestIdToken(getString(R.string.default_web_client_id)) //OAuth 2.0 client ID
+                .requestEmail()
+                .build()
+
+        // Build a GoogleSignInClient with the options specified by gso.
+        mGoogleSignInClient = GoogleSignIn.getClient(this, gso)
+        mAuth = FirebaseAuth.getInstance()
     }
 
+    public override fun onStart() {
+        super.onStart()
+        // Check if user is signed in (non-null) and update UI accordingly.
+        val currentUser = mAuth?.getCurrentUser()
+        updateUI(currentUser)
+    }
+
+    private fun signIn() {
+        val signInIntent = mGoogleSignInClient?.getSignInIntent()
+        startActivityForResult(signInIntent, RC_SIGN_IN)
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            val task = GoogleSignIn.getSignedInAccountFromIntent(data)
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                val account = task.getResult(ApiException::class.java)
+                firebaseAuthWithGoogle(account)
+            } catch (e: ApiException) {
+                // Google Sign In failed, update UI appropriately
+                Log.w(TAG, "Google sign in failed", e)
+                updateUI(null)
+            }
+
+        }
+    }
+
+    private fun firebaseAuthWithGoogle(acct: GoogleSignInAccount) {
+        Log.d(TAG, "firebaseAuthWithGoogle:" + acct.id!!)
+
+        val credential = GoogleAuthProvider.getCredential(acct.idToken, null)
+        mAuth?.signInWithCredential(credential)
+                ?.addOnCompleteListener(this) { task ->
+                    if (task.isSuccessful) {
+                        // Sign in success
+                        Log.d(TAG, "signInWithCredential:success")
+                        val user = mAuth?.getCurrentUser()
+                        updateUI(user)
+                    } else {
+                        // If sign in fails, display a message to the user.
+                        Log.w(TAG, "signInWithCredential:failure", task.exception)
+                        Snackbar.make(findViewById(R.id.activity_splash), "Authentication Failed.", Snackbar.LENGTH_SHORT).show()
+                        updateUI(null)
+                    }
+                }
+    }
+
+    private fun updateUI(user: FirebaseUser?) {
+        if (user != null) {
+            findViewById<SignInButton>(R.id.sign_in_button).setVisibility(View.GONE)
+        } else {
+            findViewById<SignInButton>(R.id.sign_in_button).setVisibility(View.VISIBLE)
+        }
+    }
+
+    override fun onClick(v: View) {
+        when (v.getId()) {
+            R.id.sign_in_button -> signIn()
+        }
```

Comment:
When you make this a lambda, feel free to axe this when statement. The id will always be `R.id.sign_in_button`.